src/llscan.cc

```diff
@@ -0,0 +1,376 @@
+    return address_byte_size_;
+  }
+
+  if (err.Fail()) {
+    return address_byte_size_;
+  }
+
+  v8::Map map(map_object);
+
+  v8::Value::InspectOptions inspect_options;
+  inspect_options.detailed = false;
+  inspect_options.print_map = false;
+  inspect_options.string_length = 0;
+
+  std::string type_name = heap_object.GetTypeName(&inspect_options, err);
+
+  /* No entry in the map, create a new one. */
+  if (mapstoinstances_.count(type_name) == 0) {
+    TypeRecord* t = new TypeRecord();
+    t->type_name = new std::string(type_name);
+    t->instances = new std::set<uint64_t>();
+    t->instances->insert(word);
+    t->instance_count = 1;
+    t->total_instance_size = map.InstanceSize(err);
+
+    mapstoinstances_.insert(std::pair<std::string, TypeRecord*>(type_name, t));
+  } else {
+    /* Update an existing instance, if we haven't seen this instance before. */
+    TypeRecord* t = mapstoinstances_.at(type_name);
+    /* Determine if this is a new instance.
+     * (We are scanning pointers to objects, we may have seen this location
+     * before.)
+     */
+    if (t->instances->count(word) == 0) {
+      t->instances->insert(word);
+      t->instance_count++;
+      t->total_instance_size += map.InstanceSize(err);
+    }
+  }
+
+  if (err.Fail()) {
+    return address_byte_size_;
+  }
+
+  found_count_++;
+
+  /* Just advance one word.
+   * (Should advance by object size, assuming objects can't overlap!)
+   */
+  return address_byte_size_;
+}
+
+
+bool FindJSObjectsVisitor::IsAHistogramType(v8::HeapObject& heap_object,
+                                            v8::Error err) {
+  int64_t type = heap_object.GetType(err);
+  v8::LLV8* v8 = heap_object.v8();
+  if (type == v8->types()->kJSObjectType) return true;
+  if (type == v8->types()->kJSArrayType) return true;
+  if (type == v8->types()->kJSTypedArrayType) return true;
+  return false;
+}
+
+
+bool LLScan::ScanHeapForObjects(lldb::SBTarget target,
+                                lldb::SBCommandReturnObject &result) {
+  /* TODO(hhellyer) - Check whether we have the SBGetMemoryRegionInfoList API
+   * available
+   * and implemented.
+   * (It may be available but not supported on this platform.)
+   * If it is then check the last scan is still valid - the process hasn't moved
+   * and we haven't changed target.
+   */
+
+  // Reload process anyway
+  process_ = target.GetProcess();
+
+  // Need to reload memory ranges (though this does assume the user has also
+  // updated
+  // RANGESFILE with data for the new dump or things won't match up).
+  if (target_ != target) {
+    MemoryRange* head = ranges_;
+    while (head != nullptr) {
+      MemoryRange* range = head;
+      head = head->next;
+      delete range;
+    }
+    ranges_ = nullptr;
+  }
+
+  /* Fall back to environment variable containing pre-parsed list of memory
+   * ranges. */
+  if (nullptr == ranges_) {
+    const char* segmentsfilename = getenv("RANGESFILE");
```

Comment:
`RANGESFILE` sounds a bit generic. What do you think about prefixing it with `LLNODE_`?